lib/api/user_token_service.rb

```diff
@@ -1,23 +1,28 @@
 module Api
   class UserTokenService
+    TYPES = %w(api ui ws).freeze
+    # Additional Requester type token ttl's for authentication
+    TYPE_TO_TTL_OVERRIDE = {'ui' => ::Settings.session.timeout}.freeze
+
     def initialize(config = Settings, args = {})
       @config = config
       @svc_options = args
-      @token_mgr = new_token_mgr(base_config[:module], base_config[:name], api_config)
     end
 
-    attr_accessor :token_mgr
-
-    # Additional Requester type token ttl's for authentication
-    #
-    REQUESTER_TTL_CONFIG = {"ui" => :ui_token_ttl}.freeze
+    def token_mgr(type)
+      @token_mgr ||= {}
+      case type
+      when 'api', 'ui' # The default API token and UI token share the same TokenStore
+        @token_mgr['api'] ||= new_token_mgr(base_config[:module], base_config[:name], api_config)
+      when 'ws'
+        @token_mgr['ws'] ||= TokenManager.new('ws', :token_ttl => ::Settings.session.timeout)
```

Comment:
What happened to the websocket specific timeout of 5 minutes ?  I thought there was a concern of this number being too big.